packages/kbn-journeys/journey/journey_config.ts

```diff
@@ -10,7 +10,16 @@ import Path from 'path';
 
 import { REPO_ROOT } from '@kbn/repo-info';
 
+import { SynthtraceGenerator } from '@kbn/apm-synthtrace-client/src/types';
+import { Readable } from 'stream';
 import { BaseStepCtx } from './journey';
+import { SynthtraceClientType } from '../services/synthtrace';
+
+interface JourneySynthtrace<T extends { '@timestamp'?: number | undefined }, O = any> {
+  type: SynthtraceClientType;
+  generator: (options: O) => Readable | SynthtraceGenerator<T>;
+  options: O;
+}
```

Comment:
Since generator might be a function without arguments passed from tests, I make options generic in efe9ed97b03d300839ace71598438a9ac471c6cb